### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Vsumskiy**

### CODE REVIEW

File: src/app/pages/auth/sign-in-form/sign-in-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  setForm(): void {
+    this.form = this.fb.group({
+      email: [null, [Validators.required, Validators.email]],
+      password: [null, [Validators.required, Validators.minLength(6)]],
+      firstName: [null, [Validators.required, Validators.minLength(2)]],
+      lastName: [null, [Validators.required, Validators.minLength(2)]]
+    });
+  }
+
+  signIn(): void {
```
Comment (iyarymovych-reenbit) [inline src/app/pages/auth/sign-in-form/sign-in-form.component.ts:45]: It's not readable method. Too much logic is written here.
Try to create a couple of methods here!

File: src/app/services/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  getAuthData(): User {
+    return JSON.parse(localStorage.getItem(this.localStorageKey));
+  }
+
+  setAuthData(user: User, remember: boolean): void {
+    this.userData = user;
+    this.isAuthenticated = true;
+    if (remember) {
+      localStorage.setItem(this.localStorageKey, JSON.stringify(user));
```
Comment (iyarymovych-reenbit) [inline src/app/services/auth.service.ts:35]: Why do you need to store whole user data in local storage?

File: src/app/app.component.ts [hunk 3]
```diff
@@ -9,11 +10,23 @@ import { JokesDataService } from './services/jokes-data.service';
...
     public jokeService: JokeService,
-    public jokesDataService: JokesDataService
+    public jokesDataService: JokesDataService,
+    private authService: AuthService
   ) {}
+
   ngOnInit(): void {
-    this.jokeService.favoritesJokes = this.jokesDataService.getDataFromLocalStorage();
+    this.authService.refreshUserData();
+    if (this.authService.isAuthenticated) {
```
[the remark this answers is not in the document]
Comment (Vsumskiy) [reply src/app/app.component.ts:22]: done.

File: src/app/interceptors/auth.interceptor.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  constructor(private authService: AuthService, private router: Router) {}
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    if (this.authService.isAuthenticated) {
+      req = req.clone({
+        setHeaders: {
+          Authorization: `Bearer ${this.authService.userData.token}`,
+          Accept: 'application/json'
```
[the remark this answers is not in the document]
Comment (Vsumskiy) [reply src/app/interceptors/auth.interceptor.ts:25]: If I don`t send this header, I get an unexpected token error :(

File: src/app/pages/auth/sign-in-form/sign-in-form.component.ts (hunk 1 above)
Comment (Vsumskiy) [reply src/app/pages/auth/sign-in-form/sign-in-form.component.ts:45]: okay

**Pull request merged**